README.md

```diff
@@ -21,86 +21,72 @@ struct MyType
+        `pair[1]`—or `:ByName`—which sorts by lexigraphical order.
+4. `FnHash(fn, [method])`: hash the result of applying `fn` to the given object. Optionally,
+   use `method` to hash the result of `fn`, otherwise calls `hash_method` on the result to
+   determine how to hash it. There are two built-in functions commonly used with
+   `FnHash`
+    - `qualified_name`: Get the qualified name of an objects type, e.g. `Base.String`
+    - `qualified_type`: Get the qualified name and type parameters of a type, e.g.
+       `Base.Vector{Int}`. 
+    Favor these functions over e.g. `string ∘ typeof` as they have been tested to provide
+    more stable values across julia verisons and sessions than the naive
+    string-ification of types.
+5. `ConstantHash(value, [method])`: hash the constant `value`. Optionally, use `method` to
+    hash the `value`, otherwise call `hash_method` on `value` to determine how to hash it.
+6. `Tuple`: apply multiple methods to hash the object, and then recursively hash their
+    results. 
 
 Your hash will be stable if the output for the given method remains the same: e.g. if
-`write` is the same for an object that uses `UseWrite`, its hash will be the same; if the
-properties are the same for `UseProperties`, the hash will be the same; etc...
-
-## Implemented methods of `hash_method`
-
-In the absence of a specific `hash_method` for your type, the following fallbacks
-are used. They are intended to avoid hash collisions as best as possible.
-
-- `Any`: 
-    - `UseWrite()` for any object `x` where `isprimitivetype(typeof(x))` is true
-    - `Use(qualified_type, UseStruct(:ByName))` for all other types
-- `NamedTuple`: `Use(qualified_name, UseStruct())`
-- `Function`: `Use("Base.Function", Use(qualified_name))`
-- `AbstractString`: `Use(qualified_name, UseWrite())`
-- `Symbol`: `Use(":", UseWrite())`
-- `String`: `UseWrite()` (note: removing the `Use(qualified_name` prevents an infinite loop)
-- `Tuple`, `Pair`: `Use(qualified_name, UseIterate())`
-- `Type`: `UseQualifiedType`
-- `AbstractArray`: `Use("Base.AbstractArray", UseSize(UseIterate()))`
-- `AbstractRange`: `Use(qualified_name, UseStruct())`
-- `AbstractSet`: `Use(qualified_name, UseTransform(sort! ∘ collect))`
-- `AbstractDict`: `Use(qualified_name, UseStruct(keys => getindex, :ByName))`
-
-There are two built-in contexts that can be used to modify these default fallbacks:
-`TablesEq` and `ViewsEq`. `TablesEq` makes any table with equivalent content have the same
-hash, and `ViewsEq` makes any array or string with the same sequence of values and the same
-size have an equal hash. You can pass one or more of these as the second argument to `stable_hash`, e.g. `stable_hash(x, ViewsEq())` or `stable_hash(x, ViewsEq(TablesEq()))`.
+`write` is the same for an object that uses `WriteHash`, its hash will be the same; if the
+fields are the same for `StructHash`, the hash will be the same; etc...
+<!-- END_HASH_TRAITS -->
```

Comment:
I think it requires the reader to understand contexts, which are not fully explained until later in the README.

Reply:
I think it would still be good to include it. This list is a convenient place to look for "what traits are available", so I think it would be good to include `HashAndContext` even if it's a little out-of-order in the README.